library/Translate/test/Test/Desugar/Sexp.hs

```diff
@@ -234,6 +256,6 @@ modLetWorkAsExpected =
         "(:defun foo ()\
         \   (let foo ()\
         \         (let bar () 3 \
-        \            (:let-type foo (() (XTZ))\
```

Comment:
Was it working before as expected?

Reply:
I'd argue no, as the form is harder to deal with in general, as the car is arguments yet the cdr is a bunch of sum types, it's just odd as is.

Reply:
Yes, I understand, the form of these aren't fully fit for purpose yet, the :module and :let-mod forms i think need a bit of tweaking. 

I'll likely have to redesign a whole lot when the syntax changes come in (the tests get the brunt of the annoying work, while the code should stay the same!)